### DISCUSSION TITLE
How to set the a font for all controls

### DISCUSSION THREAD

**Discussion by josvanouwerkerk** [AvaloniaUI/Avalonia#14466] [Q&A] [answered]
It feels that this should be fairly easy, but I'm not finding a short way to do this. I want to essentially override the main font everywhere.

What I've done:
- Added the font I want to use as avalonia resource
- Added the font as static resource in App.axaml
- Added some style selectors in App.axaml:
```
<Style Selector="TextBlock">
  <Setter Property="FontFamily" Value="{StaticResource Main}"/>
</Style>
<Style Selector="TextBox">
  <Setter Property="FontFamily" Value="{StaticResource Main}"/>
</Style>
<Style Selector="Label">
  <Setter Property="FontFamily" Value="{StaticResource Main}"/>
</Style>
<Style Selector="MenuItem">
  <Setter Property="FontFamily" Value="{StaticResource Main}"/>
</Style>
<Style Selector="RadioButton">
  <Setter Property="FontFamily" Value="{StaticResource Main}"/>
</Style>
```

This works, but it's fairly repetitive and lengthy. I've been looking for a selector that targets all controls with a FontFamily property, but I could not find it.

**Comment by kyuranger** [comment]
If all your controls are located in MainWindow, then you can directly set the FontFamily of MainWindow, and the controls in it will also be set to its FontFamily.

**Comment by josvanouwerkerk** [comment]
Well, I have multiple windows, so that would require setting it for each, but it's a decent alternative.

I do feel that it is more common to attach the font related properties to all controls and use inheritance from the parent. So then you'd also be able to set a font on any panel.

ChatGPT was suggesting selectors like *[FontFamily] or :all[FontFamily] that don't work. Did those ever work or is ChatGPT just hallucinating?

**Reply by timunie** [reply]
I think I'd check theme resources in source code and override the needed DynamicResources in App.axaml

**Comment by workgroupengineering** [comment, marked as answer]
In your app.axaml

```xaml
    <Application.Styles>
        <FluentTheme />
	<Style Selector=":is(TopLevel)">
		<Setter Property="FontFamily" Value="Comic Sans MS"/>
	</Style>
    </Application.Styles>
```
